core/trino-main/src/main/java/io/trino/sql/gen/CursorProcessorCompiler.java

```diff
@@ -254,27 +367,11 @@ private static IfStatement createProjectIfStatement(
                 .getVariable(pageBuilder)
                 .invokeVirtual(PageBuilder.class, "declarePosition", void.class);
 
-        // this.project_43(session, cursor, pageBuilder.getBlockBuilder(42)));
-        for (int projectionIndex = 0; projectionIndex < projections; projectionIndex++) {
+        for (CursorProcessorChunkClass chunkClass : chunkClasses) {
             ifStatement.ifTrue()
-                    .append(method.getThis())
-                    .getVariable(session)
-                    .getVariable(cursor);
-
-            // pageBuilder.getBlockBuilder(0)
```

Comment:
nit:  It would be better to split mechanical refactors from more complex changes. Is there a way to extract a commit with just moving this code?

Reply:
I added a new commit where I extract this loop into a method: `Extract projection invocation generation into separate method`.